src/ImageSharp/Formats/Tiff/Compression/HorizontalPredictor.cs

```diff
@@ -67,6 +67,11 @@ public static void ApplyHorizontalPrediction(Span<byte> rows, int width, int bit
         {
             ApplyHorizontalPrediction8Bit(rows, width);
         }
+        else if (bitsPerPixel == 16)
+        {
+            // Assume rows are L16 grayscale since that's currently the only way 16 bits is supported by encoder
```

Comment:
Can you clarify how you'd like to support color? Do you mean support encoding color as 16 bits per pixel, or when the user tells the encoder to use 16 bits per pixel, we should determine somehow whether to use Rgb24 or L16?

Before this change, if the user set bits per pixel in the encoder to 16, the encoder would save as Rgb24. So color at 16 bits per pixel already isn't supported. We're changing the assumption now that the user meant L16 instead of Rgb24 when using the encoder setting of 16 bits per pixel. The new implementation would only become a breaking change if the user is explicitly setting this option in the encoder which didn't do exactly as they wanted anyway.

Supporting encoding color as 16 bits per pixel would become a bigger design/architecture challenge since we'll need to figure out which pixel format the user wants to use specifically (4444, 565, 5551, La16).

Reply:
Sorry, I misread. I was thinking 16bits per component like png. 

I would assume that 16 bits per pixel can only mean luminance as you have.